File: validate_json.py
import json
import os

filepath = r"c:\Users\hp\Downloads\DOCEASE\medicament 1.json"

try:
    with open(filepath, 'r', encoding='utf-8') as f:
        data = json.load(f)
    
    full_content = "".join([p['content'] for p in data])
    
    # Save the full content to a temp file for inspection if needed
    with open("temp_full.json", "w", encoding='utf-8') as f:
        f.write(full_content)
        
    try:
        parsed = json.loads(full_content)
        print("SUCCESS: Full content is valid JSON.")
    except json.JSONDecodeError as e:
        print(f"ERROR at line {e.lineno}, col {e.colno}: {e.msg}")
        start = max(0, e.pos - 50)
        end = min(len(full_content), e.pos + 50)
        print(f"Context: ...{full_content[start:e.pos]} >>>HERE<<< {full_content[e.pos:end]}...")

except Exception as e:
    print(f"Error reading file: {e}")

File: validate_json.js
const fs = require('fs');
const path = require('path');

const filepath = 'c:\\Users\\hp\\Downloads\\DOCEASE\\medicament 1.json';

try {
    const data = JSON.parse(fs.readFileSync(filepath, 'utf8'));
    const fullContent = data.map(p => p.content).join('');

    fs.writeFileSync('temp_full.json', fullContent);

    try {
        const parsed = JSON.parse(fullContent);
        console.log("SUCCESS: Full content is valid JSON.");
    } catch (e) {
        console.error(`ERROR: ${e.message}`);
        const pos = parseInt(e.message.match(/position (\d+)/)?.[1] || 0);
        if (pos) {
            const start = Math.max(0, pos - 50);
            const end = Math.min(fullContent.length, pos + 50);
            console.log(`Context: ...${fullContent.substring(start, pos)} >>>HERE<<< ${fullContent.substring(pos, end)}...`);
        }
    }
} catch (e) {
    console.error(`Error reading file: ${e.message}`);
}
